File: backend/app/schemas/registration.py
from pydantic import BaseModel


class RegionLevelCount(BaseModel):
    region: str
    real_identity: str | None
    register_count: int
    arrive_count: int


class MatrixRow(BaseModel):
    region: str
    qianwan_register: int
    qianwan_arrive: int
    baiwan_register: int
    baiwan_arrive: int
    putong_register: int
    putong_arrive: int
    total_register: int
    total_arrive: int


class RegistrationDetail(BaseModel):
    customer_name: str | None
    sign_in_status: str | None
    customer_category: str | None
    real_identity: str | None
    attendee_role: str | None
    store_name: str | None
    region: str | None

File: frontend/src/api/registration.ts
import client, { type ApiResponse } from './client'

export interface RegionLevelCount {
  region: string
  real_identity: string | null
  register_count: number
  arrive_count: number
}

export interface MatrixRow {
  region: string
  qianwan_register: number
  qianwan_arrive: number
  baiwan_register: number
  baiwan_arrive: number
  putong_register: number
  putong_arrive: number
  total_register: number
  total_arrive: number
}

export const fetchRegistrationChart = () =>
  client.get<ApiResponse<RegionLevelCount[]>>('/v1/registration/chart').then(r => r.data.data)

export const fetchRegistrationMatrix = () =>
  client.get<ApiResponse<MatrixRow[]>>('/v1/registration/matrix').then(r => r.data.data)

export interface RegistrationDetail {
  customer_name: string | null
  sign_in_status: string | null
  customer_category: string | null
  real_identity: string | null
  attendee_role: string | null
  store_name: string | null
  region: string | null
}

export const fetchRegistrationDetail = (region?: string, level?: string) =>
  client.get<ApiResponse<RegistrationDetail[]>>('/v1/registration/detail', { params: { region, level } }).then(r => r.data.data)
